packages/shared/dao/pick-realm-algorithm/allPeers.ts

```diff
@@ -0,0 +1,41 @@
+import { Candidate } from "../types"
+import { AlgorithmContext, AlgorithmLink, AlgorithmLinkTypes, AllPeersScoreParameters } from "./types"
+import { usersCount, maxUsers, selectFirstByScore, defaultScoreAddons } from "./utils"
+
+export function allPeersScoreLink({ baseScore, discourageFillTargetPercentage, fillTargetPercentage, latencyDeductionsParameters }: AllPeersScoreParameters): AlgorithmLink {
+  function usersScore(candidate: Candidate) {
+    const count = usersCount(candidate)
+    const max = maxUsers(candidate)
+
+    // We prefer realms that have users. Those will have at least baseScore
+    if (count === 0) return 0
+
+    const linearUsersScore = (users: number) => baseScore + users
+
+    if (max) {
+      // We try to fill all realms until around the percentage provided
+      if (count >= fillTargetPercentage * max) {
+        // If this is the case, we are in the "downward" phase of the score
+        // We calculate a segment joining the fillTargetPercentage% of users with baseScore at discourageFillTargetPercentage% maxUsers
+        // In that way, when we reach discourageFillTargetPercentage% maxUsers, realms that have at least one user start to get prioritized
+        const segment = { a: { x: fillTargetPercentage * max, y: linearUsersScore(fillTargetPercentage * max) }, b: { x: discourageFillTargetPercentage * max, y: baseScore } }
```

Comment:
can we rename `a` and `b` to `startingPoint` `endingPoint` or something a bit more descriptive?

Reply:
done